File: Data Structures & Algorithms/majority-element/submission-1.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        count = 0
        res = -1
        for i in nums:
            if count == 0: res = -1;
            if res == -1 :
                res = i
                count = 1
                continue
            if i != res:
                count-=1
                
            else :
                count+=1
        return res 

File: Data Structures & Algorithms/anagram-groups/submission-1.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string,vector<string>>mp;
        for(string str:strs){
            string str_copy = str;
            sort(str_copy.begin(),str_copy.end());
            mp[str_copy].push_back(str);
        }
        vector<vector<string>>res(mp.size());
        int ct = 0;
        for (auto &[a,vec]:mp){
            for (string st:vec){
                res[ct].push_back(st);
            }
            ct++;
        }
        return res;
    }
};
